utils/resource-monitor/graph_pidstat.py

```diff
@@ -0,0 +1,215 @@
+    type=str, 
+    required=True,
+    help='Provide the number of cores to use in the batch job.')
+
+parser.add_argument(
+    '--dataDescription', 
+    type=str, 
+    required=True,
+    help='Provide a description of the data.')
+
+parser.add_argument('--outputParquetPath', 
+    type=str, 
+    default='/tmp/hapi-test/',
+    help='Provide the output path for the parquet files generated by the batch job.')
+
+parser.add_argument('--outputResultsPath', 
+    type=str, 
+    default='/tmp/hapi-performance/',
+    help='Provide the output path for the graphs and csv files.')
+
+def get_machine_mem():
+    """
+    Gets the RAM of the local machine.
+
+    Args:
+      None
+
+    Returns:
+      Integer representation of the local machine RAM
+    """
+    mem_info = subprocess.run(['grep', 'MemTotal', '/proc/meminfo'], stdout=subprocess.PIPE).stdout.split()
+    return int(int(mem_info[1])/1000000)
+
+def get_machine_cpu():
+    """
+    Gets the CPU/number of cores of the local machine.
+
+    Args:
+      None
+
+    Returns:
+      Integer representation of the number of cores of the local machine
+    """
+    cpu_info = subprocess.run(['nproc'], stdout=subprocess.PIPE).stdout
+    return int(cpu_info)
+
+def monitor_pipeline_pidstat(num_proc, data_description, output_parquet_path, output_results_path):
+    """
+    Creates output subdirs and executes bash pipeline monitoring script.
+
+    Args:
+      num_proc: Number of processes the batch job is set to run with
+      data_description: Description of the dataset
+      output_parquet_path: Directory of the output parquet files
+      output_results_path: Directory of the output csv files and graphs
+
+    Returns:
+      None
+    """
+    if not os.path.exists(output_results_path + str(data_description) + '/graphs'): os.makedirs(output_results_path + str(data_description) + '/graphs')
+    if not os.path.exists(output_results_path + str(data_description) + '/tables'): os.makedirs(output_results_path + str(data_description) + '/tables')
+    if not os.path.exists(output_results_path + str(data_description) + '/raw'): os.makedirs(output_results_path + str(data_description) + '/raw')
+    subprocess.call(['sh', './monitor_pipeline.sh', num_proc, data_description, output_parquet_path, output_results_path])
+
+def make_pidstat_tables(file_name, type):
+    """
+    Creates pandas tables from raw pidstat data.
+
+    Args:
+      file_name: Name of the .txt file containing raw pidstat output
+      type: Type of the resource relevant to the table (Pipeline, Server, or DB)
+
+    Returns:
+      Pandas table with CPU, memory and I/O usage info for the resource.
+    """
+
+    #Parse the raw .txt file into a pandas dataframe
+    total_usage = []
+    seconds_elapsed = 0
+    with open(file_name) as f:
+        for line in f:
+            usage = []
+            if line[0].isdigit():
+                usage = line.split()
+
+                if type == 'DB ' and usage[-3] != 'hapi':
```

Comment:
Please add some comments about expected format of each line, either here or in a single location before the loop.